Add a simple VERSIONING.md to the repo (#455)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/VERSIONING.md

# Versioning and Stability

> As per OpenTelemetry specification this repository uses [Semantic Versioning 2.0.0](https://semver.org/).

Reviewer: ```suggestion
As per the OpenTelemetry specification this repository uses [Semantic Versioning 2.0.0](https://semver.org/).
```

> This means that all artifacts have a version of the format `MAJOR.MINOR.PATCH` or `MAJOR.MINOR.PATCH-<pre-release-id>`. For detailed information about version and stability for OpenTelemetry see

Reviewer: ```suggestion
This means that the version of all artifacts is in the format `MAJOR.MINOR.PATCH` or `MAJOR.MINOR.PATCH-<pre-release-id>`. For detailed information about version and stability for OpenTelemetry see
```

[Versioning and stability for OpenTelemetry clients](https://github.com/open-telemetry/opentelemetry-specification/blob/bccdb63e3da14d7eb3a4f3090e270126373069ba/specification/versioning-and-stability.md#versioning-and-stability-for-opentelemetry-clients).

> In this repository `MAJOR` and `MINOR` version numbers are going to match the [OpenTelemetry .NET SDK](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#opentelemetry-net-sdk)
> currently shipped with the artifacts produced by CI.

Reviewer: ```suggestion
In this repository, `MAJOR` and `MINOR` version numbers match the version numbers of the
[OpenTelemetry .NET SDK](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#opentelemetry-net-sdk) shipped with the artifacts produced by CI.
```

> As auto-instrumentation will inject assemblies during application runtime
> conflicts between assemblies are possible and may demand the workarounds
> listed at [Handling of assembly version conflicts](./troubleshooting.md/#handling-of-assembly-version-conflicts).

Reviewer: ```suggestion
As the auto-instrumentation injects assemblies at application runtime,
conflicts between assemblies might happen and might require workarounds.
See [Handling of assembly version conflicts](./troubleshooting.md/#handling-of-assembly-version-conflicts) for more information.
```
